libfairygui/Classes/display/FUIInput.cpp

```diff
@@ -78,7 +78,7 @@ void FUIInput::setKeyboardType(int value)
 
 void FUIInput::openKeyboard()
 {
-    touchDownAction(this, cocos2d::ui::Widget::TouchEventType::ENDED);
+    EditBox::openKeyboard();

```

Comment:
EditBox::openKeyboard is not exists?

Reply:
touchDownAction is deprecated.
EditBox [[
/**
         * @js NA
         * @lua NA
         * @deprecated Use openKeyboard() instead to open the keyboard
         */
        CC_DEPRECATED_ATTRIBUTE
        void touchDownAction(Ref *sender, TouchEventType controlEvent);

        void openKeyboard() const;
]]

Reply:
OK. The API changes in the latest version 3.17.1.